### ISSUE TITLE
[checkoutlinesufo] Don’t create hash if there is no processed glyphs layer

### ISSUE THREAD

**Issue by frankrolf** [adobe-type-tools/afdko#284]
For a UFO on which `checkoutlinesufo` has never been run, the command
`checkoutlinesufo -w -e font.ufo` creates a hash map.
The follow-up command `makeotf -f font.ufo` results in

```
src glyph A is newer than processed layer. programName: CheckOutlines
Done.
```

This is confusing because no processed glyph layer exists.
I propose omitting the creation of the hash map if changes are written in the foreground layer.

**Comment by miguelsousa** [comment]
That message happens because of #349.

> I propose omitting the creation of the hash map if changes are written in the foreground layer.

I disagree. The goal of the hash file is to speed up subsequent processes. Just because the changes are being written to the default layer shouldn't preclude keeping a tally of that layer. Because if you later on add or modify glyphs, checkoulinesufo can quickly skip over that ones for which the hash matches the default layer.

**Comment by cjchapman** [comment]
So,  it seems to me that there’s a fundamental problem with the way CheckOutlinesUFO and ufoTools currently interact, which is that ufoTools is doing all its hash calculations based on data in the UFO file on the disk, and changes to that UFO file don’t happen until CheckOutlinesUFO finishes its work and writes out the UFO file.

**Comment by frankrolf** [comment]
Christopher is getting to the root of the problem. I was thinking about this more – of course it still makes sense to create a hash for the foreground layer. 
Instead of omitting the hash, perhaps it needs to store the info if a processed layer actually exists or not.

**Comment by miguelsousa** [comment]
**checkoutlinesufo** saves glyph hashes by default, no matter if the option `-w` (i.e. write any changes to the UFO's default layer) is used or not. I don't think that approach should be changed.
If you don't want the hashmap file to be written, you'll need use the `--clear-hash-map` option.

As for the issue with **makeotf**, it got fixed by #694.
